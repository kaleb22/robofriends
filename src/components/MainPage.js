import React, { Component } from 'react';
import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll';
import ErrorBoundry from '../components/ErrorBoundry';
import '../containers/App.css'

class MainPage extends Component {

  componentDidMount() {
    this.props.onRequestRobots();
  }

  filteredRobots = () => {
    return this.props.robots.filter(robot => {
      return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
    });
  }

  render() {
    const { onSearchChange, isPending } = this.props;
    
    return isPending ?
      <h1>Loading...</h1> :
    (
      <div className='tc'>
        <h1 className='f1'>Robofriends</h1>
        <SearchBox searchChange={onSearchChange}/>
        <Scroll>
          <ErrorBoundry>
            <CardList robots={this.filteredRobots()}/>
          </ErrorBoundry>
        </Scroll>
      </div>
    );
  }
}

export default MainPage;