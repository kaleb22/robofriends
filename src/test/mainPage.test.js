import { shallow} from 'enzyme'
import React from 'react'
import MainPage from '../components/MainPage'

let wrapper;

beforeEach( () => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: '',
    isPending: false
  }

  wrapper = shallow(<MainPage { ...mockProps }/>);
})

it('renders MainPage component', () => {
  expect(wrapper).toMatchSnapshot();
});

it(' filters robots correctly', () => {
  const mockProps2 = {
    onRequestRobots: jest.fn(),
    robots: [{
      id: 3,
      name: 'John Doe',
      email: 'John@gmail.com'
    }],
    searchField: 'John',
    isPending: true
  }

  const wrapper2 = shallow(<MainPage { ...mockProps2 }/>);

  expect(wrapper.instance().filteredRobots()).toEqual([]);
  expect(wrapper2.instance().filteredRobots()).toEqual([{
    id: 3,
    name: 'John Doe',
    email: 'John@gmail.com'
  }]);
});


