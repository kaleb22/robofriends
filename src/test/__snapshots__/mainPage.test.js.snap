// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render MainPage component 1`] = `ShallowWrapper {}`;

exports[`expects to render MainPage component 1`] = `ShallowWrapper {}`;

exports[`renders MainPage component 1`] = `ShallowWrapper {}`;
